### PR TITLE
SWATCH-1717 & SWATCH-1718: Create metering-rhel and metering-rhel-job profiles, and define ClowdApps

### PR DESCRIPTION
**Pull request by lindseyburnett**
Jira issue: [SWATCH-1718](https://issues.redhat.com/browse/SWATCH-1718)

## Description

Defining a new ClowdApp to deploy another instance of swatch-metrics, named swatch-metrics-rhel, that connects to RHEL telemeter instance instead of openshift telemeter instance.  There is no need for a separate purge events job, so I omitted it from the swatch-metrics-rhel definition.

This includes creating separate profiles.  This is necessary to allow each deployment to bind to its dedicated kafka topic, and enable the deployments to scale independently.

The part that "points" to the rhel telemeter instance is actually environment variables on the token-refresher sidecar.  These environment variables are set via vault secrets.

## Testing

What we're looking to verify here is that the `swatch-metrics-rhel-*` pods deploy, start up using the appropriate spring profile, connect to the appropriate topic, and the token refresher container uses the swatch-metrics-rhel secret.

The only other thing we can try for verification is hitting the internal endpoint for metrics sync and confirm there are no http errors in the log.  Until there is a product configured in swatch that has data available regularly available in rhel telemeter, we won't actually be able to see swatch consume anything and save it to the database.

## checkout this branch
```bash
git checkout lburnett/metering-rhel-profiles
```

## reserve namespace & enable token refresher sidecar

```bash
export NAMESPACE=$(bonfire namespace reserve -f)

```

```
oc patch env env-$NAMESPACE -p '{"spec":{"providers":{"sidecars":{"tokenRefresher":{"enabled":true}}}}}' --type=merge
```

## deploy swatch-metrics component to namespace

Note about terminology:  "swatch-metrics" is still the name of the component, as defined by our resourceTemplate entry in app-interface.  Up until this point, our clowdapps & components were 1 to 1 in regards to naming.  Now, our "swatch-metrics" component contains two clowdapps - swatch-metrics and swatch-metrics-rhel.  Eventually (out of scope), the swatch-metrics components will be named "swatch-metrics-openshift" and "swatch-metrics-rhel".

This command assumes that the PR has already built and pushed a image to quay for this githash.

```bash
VER=$(git rev-parse --short=7 HEAD); bonfire deploy -n $NAMESPACE rhsm --optional-deps-method=none \
-C swatch-metrics \
-C swatch-tally \
--no-remove-resources=all \
--no-release-on-fail \
-i quay.io/cloudservices/rhsm-subscriptions=pr-2557-$VER \
-p swatch-metrics/IMAGE=quay.io/cloudservices/rhsm-subscriptions \
-p swatch-tally/IMAGE=quay.io/cloudservices/rhsm-subscriptions
```

## verification

EE have the following pods deployed successfully:

* swatch-metrics-service
* swatch-metrics-sync
* swatch-metrics-purge-events
* swatch-metrics-rhel-service
* swatch-metrics-rhel-sync

What we're looking for in each pod

**swatch-metrics-service**
* has a token refresher container.  its environment variables list `swatch-metrics-token-refresher` secrets still
* application logs say `The following 2 profiles are active: "openshift-metering-worker", "kafka-queue"`
* application log says `Initializing metering manager. Topic: platform.rhsm-subscriptions.metering-tasks`

**swatch-metrics-sync**
tip: if there isn't a pod yet (because scheduling), you can make one manually with 
`oc create job --from=cronjob/swatch-metrics-sync swatch-metrics-sync-manual`

* application log says `The following 2 profiles are active: "metering-job", "kafka-queue"`
* application log says `Initializing metering manager. Topic: platform.rhsm-subscriptions.metering-tasks`
* no http errors in application log

**swatch-metrics-rhel-service**
* has a token refresher container.  its environment variables list `swatch-metrics-rhel-token-refresher` secrets
* application logs say `The following 2 profiles are active: "metrics-rhel", "kafka-queue"`
* application logs say `Initializing metering manager. Topic: platform.rhsm-subscriptions.metering-rhel-tasks`

**swatch-metrics-rhel-sync**
tip: if there isn't a pod yet (because scheduling), you can make one manually with 
`oc create job --from=cronjob/swatch-metrics-rhel-sync swatch-metrics-rhel-sync-manual`

* application logs say `The following 2 profiles are active: "metrics-rhel-job", "kafka-queue"`
* application logs say `Initializing metering manager. Topic: platform.rhsm-subscriptions.metering-rhel-tasks`
* no http errors in logs

### CODE REVIEW

**Comment by kartikshahc** [comment]
Verified all steps in EE: ephemeral-jiwbqx and it looks good. Checking on couple of more things in code.

File: src/main/resources/application-metrics-rhel.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+METERING_RHEL_TASK_TOPIC: ${clowder.kafka.topics[?(@.requestedName == 'platform.rhsm-subscriptions.metering-rhel-tasks')].name:platform.rhsm-subscriptions.metering-rhel-tasks}
+SERVICE_INSTANCE_INGRESS_TOPIC: ${clowder.kafka.topics[?(@.requestedName == 'platform.rhsm-subscriptions.service-instance-ingress')].name:platform.rhsm-subscriptions.service-instance-ingress}
```
Comment (kartikshahc) [inline src/main/resources/application-metrics-rhel.yaml:2]: For my reference:
METERING_RHEL_TASK_TOPIC is for incoming 
SERVICE_INSTANCE_INGRESS_TOPIC is for outgoing events

**Review by kartikshahc — APPROVED** [review]
LGTM 
The querytemplates for Prometheus will remain the same for two different processes for now until they're producing data to rhelemeter to test.

**Comment by himmAllRight** [comment]
I ran this in EE and also quickly checked in stage now that it's merged, LGTM.

**Pull request merged**